neural_modelling/src/neuron/population_table/population_table_binary_search_impl.c

```diff
@@ -169,14 +171,17 @@ bool population_table_get_first_address(
             }
 
             last_neuron_id = get_neuron_id(entry, spike);
+            last_spike = spike;
             next_item = entry.start;
             items_to_go = entry.count;
 
             log_debug("spike = %08x, entry_index = %u, start = %u, count = %u",
                     spike, imid, next_item, items_to_go);
 
+            uint32_t local_spike_id;
             return population_table_get_next_address(
-                    row_address, n_bytes_to_transfer);
+                    &local_spike_id, row_address, n_bytes_to_transfer,
+                    neuron_id);
```

Comment:
This is an anomaly caused by the fact that this calls through to the "next" method which does most of the rest of the functionality.  The issue is that the "first" method takes the spike id in, whereas the "next" method returns it out.  I was unsure whether it was possible to pass the address of an argument (and thought it might have side effects) so I did this instead.

Reply:
Comment added